### PR TITLE
executor: fix panic when execute change pump state

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
when execute sql like `change pump to node_state ='paused' for node_id 'pump1'`, tidb will return error `connection running loop panic`

### What is changed and how it works?
fix it

### Check List 

Tests 

 - Manual test (exec sql)

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -188,9 +188,11 @@ func Next(ctx context.Context, e Executor, req *chunk.Chunk) error {
 		start := time.Now()
 		defer func() { base.runtimeStats.Record(time.Since(start), req.NumRows()) }()
 	}
-	sessVars := base.ctx.GetSessionVars()
-	if atomic.CompareAndSwapUint32(&sessVars.Killed, 1, 0) {
-		return ErrQueryInterrupted
+	if base.ctx != nil {
```
Comment (suzaku) [inline executor/executor.go]: When would `base.ctx` be `nil`?

Comment (WangXiangUSTC) [reply executor/executor.go]: maybe some other `Executor`  forget to initial `baseExecutor` just like `ChangeExec `

Comment (suzaku) [reply executor/executor.go]: Shall we make it an error to leave `baseExecutor` out? It seems if it's `nil` then we are introducing something that cannot be interrupted by `KILL` here.

Comment (WangXiangUSTC) [reply executor/executor.go]: removed, PTAL again

Comment (suzaku) [reply executor/executor.go]: How about adding a test for this case?

**Comment by shenli** [comment]
Please add a unit test case.

**Comment by WangXiangUSTC** [comment]
@shenli @suzaku add a unit test, PTAL again

**Comment by jackysp** [comment]
Please fix CI, @WangXiangUSTC .

**Pull request merged**